src/sentry/search/eap/spans.py

```diff
@@ -310,6 +310,61 @@ def _resolve_having(
             return parsed_terms[0], resolved_contexts
         return None, []
 
+    def _handle_virtual_column_terms(
+        self,
+        term: event_search.SearchFilter,
+        resolved_column: ResolvedColumn,
+        context: VirtualColumnContext | None,
+    ) -> tuple[ResolvedColumn, float | datetime | Sequence[float] | Sequence[str]]:
+        raw_value = term.value.raw_value
+        if context is None or isinstance(raw_value, datetime):
+            return resolved_column, raw_value
+        raw_value = cast(str | int | list[str], raw_value)
+
+        # Convert the term to the expected values
+        final_raw_value: str | int | list[str] | list[int] = []
+        reversed_context = {v: k for k, v in context.value_map.items()}
+        if term.value.is_wildcard():
+            # Avoiding this for now, but we could theoretically do a wildcard search on the resolved contexts
+            raise InvalidSearchQuery(f"Cannot use wildcards with {term.key}")
+        elif isinstance(raw_value, list):
+            new_value = []
+            for raw_iterable in raw_value:
+                if term.key.name not in constants.PROJECT_FIELDS and raw_iterable == "Unknown":
```

Comment:
What columns can support `Unknown` as a value? Should this assumption be specified somewhere for all virtual column contexts? Or is this a `device.class` specific thing?

Reply:
specific to device.class, good callout i'll change the first condition to specifically be for the current column